modules/cluster-logging-elasticsearch-retention.adoc

```diff
@@ -0,0 +1,123 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-elasticsearch.adoc
+
+[id="cluster-logging-elasticsearch-retention_{context}"]
+= Configuring log retention time
+
+You can specify how long the default Elasticsearch log store keeps indices 
+using a separate _retention policy_ for each of the three log sources: 
+infrastructure logs, application logs, and audit logs. The retention policy, 
+configured using the `maxAge` parameter in the Cluster Logging Custom Resource 
+(CR), determines when to delete the Elasticsearch indices. 
+
+If you do not create a retention policy, logs are retained for seven days by
+default. If you do not specify a retention policy for all three log sources,
+logs from only the sources with a retention policy are stored. For example, 
+if you do not set a retention policy for audit logs, the audit logs will not
+be retained and there will be no *audit-* index in Elasticsearch or Kibana. 
+
+.Prerequisites
+
+* Cluster logging and Elasticsearch must be installed.
+
+.Procedure
+
+To configure the log retention time:
+
+. Edit the Cluster Logging CR to add or modify the `retentionPolicy` parameter:
++
+----
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+...
+spec:
+  managementState: "Managed"
+  logStore:
+    type: "elasticsearch"
+    elasticsearch:
+      nodeCount: 3
+    retentionPolicy: <1>
+      application:
+        maxAge: 1d
+      infra:
+        maxAge: 7d
+      audit:
+        maxAge: 7d
+...
+----
+<1> Specify the time that Elasticsearch should retain each log source. Enter an 
+integer and a time designation: weeks(w), hours(h/H), minutes(m) and seconds(s). 
+For example, `1d` for one day. Logs older than the `maxAge` are deleted. 
+By default, logs are retained for seven days. 
+
+. You can verify the settings in the Elasticsearch Custom Resource (CR).
++
+For example, the following Elasticsearch CR has a retention policy to roll over 
+active indices for the infrastructure logs every eight hours and the old indices 
```

Comment:
not sure how exactly to express this but the criteria for rollover is when and index meets one of several conditions:
* age
* size
* number of documents

the index will be rolled if any of those conditions are exceeded.  It could be the index is significantly less then 8 hours but  its size in Gb is met

Reply:
Per @QiaolingTang 

> the rollover will happen when the index matches any one of:
>    * the index has been a write_index for more than maxAge
>    * the index size > 40gb * number_of_primary_shards
>    * the index doc count > 40960 * number_of_primary_shards
 

https://github.com/openshift/openshift-docs/pull/22403#discussion_r437078356

Reply:
Are we exposing anything else than the age?

Reply:
@sichvoge In the ClusterLogging, only the age and the log type.

@mburke5678 the `maxAge` I said is the value of `maxAge` in the Elasticsearch, not the value in the ClusterLogging.